src/listings/data_structures.md

```diff
@@ -2,197 +2,8 @@
-
-The number of election proofs a miner can generate in a given epoch will determine the block reward it earns.
-
-# Message
-
-`Message` data structures in Filecoin describe operations that can be performed on the Filecoin VM state
-(e.g., FIL transactions between accounts).
-To facilitate the process of producing secure protocol implementations,
-we explicitly distinguish between
-{{<sref crypto_signatures "signed and unsigned">}} `Message` structures.
-
-{{<goFile Message>}}
-{{<goFile UnsignedMessage>}}
-{{<goFile SignedMessage>}}
-{{<goFile MessageReceipt>}}
-
-
-# State Tree
-
-The state tree keeps track of the entire state of the {{<sref vm>}} at any given point.
-It is a map from `Address` structures to `Actor` structures, where each `Actor`
-may also contain some additional `ActorState` that is specific to a given actor
-type.
-
-{{<goFile StateTree>}}
-
-
-# Actor
-
-{{<goFile Actor>}}
-
-
-# Signature
-
-{{<sref crypto_signatures "Cryptographic signatures">}} in Filecoin are represented
-as byte arrays, and come with a tag that signifies what key type was used to create
-the signature.
-
-{{<goFile Signature>}}
-
-
-# FaultSet
-
-`FaultSet` data structures are used to denote which sectors failed at which block heights.
-
-{{<goFile FaultSet>}}
-
-In order to make the serialization more compact,
-the `index` field denotes a block height offset from the start of the corresponding
-miner's proving period.
-
-
 # Basic Types
 
-## CID
-For most objects referenced by Filecoin, a Content Identifier (CID for short) is used.
-This is effectively a hash value, prefixed with its hash function (multihash)
-as well as extra labels to inform applications about how to deserialize the given data.
-For a more detailed specification, we refer the reader to the
-[IPLD repository](https://github.com/ipld/cid).
-
-
-## Timestamp
-{{<goFile Timestamp>}}
-
-
-## PublicKey
-The public key type is simply an array of bytes.
-{{<goFile PublicKey>}}
-
-
-## BytesAmount
-BytesAmount is just a re-typed Integer.
-{{<goFile BytesAmount>}}
-
-
-## PeerId
-The serialized bytes of a libp2p peer ID.
-{{% todo %}} Spec incomplete; take a look at [this PR](https://github.com/libp2p/specs/pull/100).{{% /todo %}}
-{{<goFile PeerId>}}
-
-
-## Bitfield
-Bitfields are a set encoded using a custom run length encoding: RLE+.
-{{<goFile Bitfield>}}
-
-
-## SectorSet
-A sector set stores a mapping of sector IDs to the respective `commR`s.
-{{<goFile SectorSet>}}
-
-{{% todo %}}
-Improve on this; see https://github.com/filecoin-project/specs/issues/116.
-{{% /todo %}}
-
-
-## SealProof
-SealProof is an opaque, dynamically-sized array of bytes.
-{{<goFile SealProof>}}
-
-
-## PoSTProof
-PoSTProof is an opaque, dynamically-sized array of bytes.
-{{<goFile PoSTProof>}}
-
-
-## TokenAmount
-A type to represent an amount of Filecoin tokens.
-{{<goFile TokenAmount>}}
-
-
-## SectorID
-Uniquely identifies a miner's sector.
-{{<goFile SectorID>}}
-
-
 # RLE+ Bitset Encoding
```

Comment:
@anorth @jzimmerman I left the following below because I do not know whether still relevant to the protocol and/or captured elsewhere in the spec (when needed). If you know for a fact that some of these are stale, please remove before merge.

Reply:
We use RLE+ bitfields quite extensively and it looks right but not sure about the rest. The listing section is supposed to link concepts and terms back to different parts of the spec where they are defined. We can remove most of them here and populate this section later when the spec stabilizes.

Reply:
will leave as is for now then unless I hear otherwise.